### DISCUSSION TITLE
Issues with Sanity Integration on GitHub and Vercel – Need Help Resolving

### DISCUSSION THREAD

**Discussion by medlemine-medelmoctar** [community/community#137527] [Programming Help]
### Body

I'm running into problems with my Sanity project when deploying through GitHub and Vercel. Everything works fine locally, but when I push to GitHub and Vercel tries to deploy, the deployment fails or the Sanity features don't work as expected. Has anyone experienced similar issues? What could be causing this, and how can I troubleshoot and resolve the problem? Any advice or steps to fix this would be greatly appreciated!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by iyehah** [comment]
# Resolving Issues with Sanity Integration on GitHub and Vercel

If you're facing issues with integrating Sanity when deploying through GitHub and Vercel, follow these steps to troubleshoot and resolve the problem:

## 1. Check Deployment Logs

Review the deployment logs on Vercel to identify any errors or warnings related to Sanity. This can provide useful insights into what might be causing the issue.

## 2. Verify Environment Variables

Ensure that all necessary environment variables for Sanity are correctly set in your Vercel project settings. Check for variables such as:
- `SANITY_PROJECT_ID`
- `SANITY_DATASET`

These should match the configuration in your Sanity project.

## 3. Review Sanity Configuration

Double-check your Sanity client configuration in your application. Make sure the `projectId`, `dataset`, and any other required settings are properly configured.

## 4. Update Dependencies

Ensure that you have the latest versions of Sanity-related dependencies in your project. Outdated packages can sometimes cause issues. Update these packages using:

```bash
npm update @sanity/client @sanity/desk-tool

**Reply by medlemine-medelmoctar** [reply]
`npm update @sanity/client @sanity/desk-tool` This solves the problem.

**Comment by ebndev** [comment]
Please see [this comment](https://github.com/orgs/community/discussions/137529#discussioncomment-10549789) for reference to why this discussion was closed and locked. Thanks!
